utils/senslog.c

```diff
@@ -0,0 +1,249 @@
+/*
+ * Phoenix-Pilot
+ *
+ * pilot-core utilities
+ * 
+ * sensor logging utility
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-Pilot software
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <sys/time.h>
+
+#include <sensc.h>
+
+struct {
+	time_t diffMax; /* Maximum time span on logging process (in microseconds) */
+	time_t step;    /* Step between logs (in microseconds) */
+	bool raw;       /* use corrections in measurements */
+
+	/* device log selection flags */
+	struct {
+		bool accel;
+		bool gyro;
+		bool mag;
+		bool baro;
+	} flags;
+} senslog_common;
+
+
+static void senslog_help(const char *progname)
+{
+	printf("Usage: %s [options]\n options:\n", progname);
+	printf("  -t,\tlogging time in full seconds\n");
+	printf("  -s,\tsampling interval in milliseconds\n");
+	printf("  -d [agmb]\tdevices to be logged as string\n");
+	printf("     a - accelerometer\n");
+	printf("     g - gyroscope\n");
+	printf("     m - magnetometer\n");
+	printf("     b - barometer\n");
+	printf("  -r raw mode, don`t use calib.conf corrections\n");
+}
+
+/* Parse devices log selection options */
+static int senslog_parseDevices(const char *devs)
+{
+	bool err = false;
+
+	if (devs == NULL) {
+		return -1;
+	}
```

Comment:
Applying this change, you will need to check only an `err` value in `95` line.
```suggestion
	if (devs == NULL || strlen(devs) == 0) {
		return -1;
	}
```

Reply:
Good idea! I have applied this check as separate if clause just below the highlighted one to be more secure (to not rely with condition output on conditions order)